src/lib.rs

```diff
@@ -0,0 +1,364 @@
+        self.inner.stdout(cfg);
+        self
+    }
+
+    /// Forwards to [`std::process::Command::stderr`][]
+    pub fn stderr<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Self {
+        self.inner.stderr(cfg);
+        self
+    }
+
+    /// Forwards to [`std::process::Command::get_program`][]
+    pub fn get_program(&self) -> &OsStr {
+        self.inner.get_program()
+    }
+
+    /// Forwards to [`std::process::Command::get_args`][]
+    pub fn get_args(&self) -> CommandArgs<'_> {
+        self.inner.get_args()
+    }
+
+    /// Forwards to [`std::process::Command::get_envs`][]
+    pub fn get_envs(&self) -> CommandEnvs<'_> {
+        self.inner.get_envs()
+    }
+
+    /// Forwards to [`std::process::Command::get_current_dir`][]
+    pub fn get_current_dir(&self) -> Option<&Path> {
+        self.inner.get_current_dir()
+    }
+}
+
+/// Bafflingly, tracing provides no builtin way to log with a dynamic log-level
+/// (it's really really obsessed with being able to compile and cache log levels),
+/// so here's a macro that does that.
+macro_rules! log {
+    ($lvl:expr, $fmt:expr, $($arg:tt)*) => {
+        match $lvl {
+            tracing::Level::TRACE => {
+                tracing::trace!($fmt, $($arg)*);
+            }
+            tracing::Level::DEBUG => {
+                tracing::debug!($fmt, $($arg)*);
+            }
+            tracing::Level::INFO => {
+                tracing::info!($fmt, $($arg)*);
+            }
+            tracing::Level::WARN => {
+                tracing::warn!($fmt, $($arg)*);
+            }
+            tracing::Level::ERROR => {
+                tracing::error!($fmt, $($arg)*);
+            }
+        }
+    }
+}
+
+/// Diagnostic APIs (used internally, but available for yourself)
+impl Cmd {
+    /// Check `Status::success`, producing a contextful Error if it's `false`.`
+    pub fn check_status(&self, status: ExitStatus) -> Result<()> {
+        if status.success() {
+            Ok(())
+        } else {
+            Err(AxoprocessError::Status {
+                summary: self.summary.clone(),
+                status,
+            })
+        }
+    }
+
+    /// Invoke [`Cmd::check_status`][] if [`Cmd::check`][] is `true`
+    /// (defaults to `true`).
+    pub fn maybe_check_status(&self, status: ExitStatus) -> Result<()> {
+        if self.check_status {
+            self.check_status(status)?;
+        }
+        Ok(())
+    }
+
+    /// Log the current Command using the method specified by [`Cmd::log`][]
+    /// (defaults to [`tracing::info!`][]).
+    pub fn log_command(&self) {
+        let Some(strategy) = self.log else {
+            return;
+        };
+        match strategy {
+            LogStrategy::Stdout => {
+                println!("exec {:?}", self.inner)
+            }
+            LogStrategy::Stderr => {
+                eprintln!("exec {:?}", self.inner)
+            }
+            LogStrategy::Tracing(level) => {
+                log!(level, "exec {:?}", self.inner);
```

Comment:
this is probably uglier formatting than we have to use?

Reply:
also  we could use the command's `description` here, but some conflict with other comments about the "failed to" prefix.